grabbit/src/test/groovy/com/twcable/grabbit/jcr/ProtoNodeDecoratorSpec.groovy

```diff
@@ -0,0 +1,127 @@
+package com.twcable.grabbit.jcr
+
+import com.twcable.grabbit.proto.NodeProtos
+import com.twcable.grabbit.proto.NodeProtos.Node as ProtoNode
+import spock.lang.Specification
+
+import javax.jcr.Node
+import javax.jcr.Session
+
+import static javax.jcr.PropertyType.STRING
+import static org.apache.jackrabbit.JcrConstants.JCR_MIXINTYPES
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+@SuppressWarnings("GroovyAccessibility")
+class ProtoNodeDecoratorSpec extends Specification {
+
+    ProtoNode decoratedProtoNode
+
+
+    def setup() {
```

Comment:
``` decoratedProtoNode ``` is used in most of the features.  I'd like to use setup() to keep potential side-effects isolated.    Yeah, it's not used for "ProtoNodeDecorator can not be constructed with a null ProtoNode", but meh...the alternative doesn't read much better, and extra setup time I think is minimal.  Happy to change if you disagree.